### PR TITLE
sshfs_mount: redirect STDERR streams when spawning

### LINKED ISSUE #27: Error stream is never closed
In devstudio we do "vagrant up" for cdk2.1-rc3/cdk/components/rhel/rhel-ose Vagrant file with vagrant-sshfs plugin installed.
Then we read output and error streams. Output stream is terminated eventually but error stream is always open. We do read "Warning: Permanently added '[127.0.0.1]:2222' (ECDSA) to the list of known hosts." from the error stream which looks like came from sshfs plugin.
devstudio is stuck because the error stream is never closed.

### PR DESCRIPTION
**Pull request by dustymabe**
Otherwise they stay open and some "wrapping" processes can't handle
that. Fixes #27

### CODE REVIEW

**Comment by dustymabe** [comment]
Tested on Linux system as well as Windows 7 + cygwin.

@LalatenduMohanty @robstryker please confirm testing results.

**Comment by LalatenduMohanty** [comment]
Tested on OS X and works fine. Both success and failure condition.

**Comment by jeffmaury** [comment]
Version is still 1.1.0. Aren't we facing update problems ?

File: lib/vagrant-sshfs/cap/linux/sshfs_mount.rb [hunk 1]
```diff
@@ -142,6 +143,12 @@ def self.sshfs_slave_mount(machine, opts, hostpath, expanded_guest_path)
           r1, w1 = IO.pipe # reader/writer from pipe1
           r2, w2 = IO.pipe # reader/writer from pipe2
 
+          # For issue #27 we'll need to create a tmp files for STDERR
```
Comment (hferentschik) [inline lib/vagrant-sshfs/cap/linux/sshfs_mount.rb]: why is a temp file needed? I don't quite understand how this fixes the issue.

File: lib/vagrant-sshfs/cap/linux/sshfs_mount.rb [hunk 2]
```diff
@@ -157,8 +164,8 @@ def self.sshfs_slave_mount(machine, opts, hostpath, expanded_guest_path)
           #          stdin <= r2        pipe2         w2 <= stdout 
           #
           # Wire up things appropriately and start up the processes
-          p1 = spawn(sftp_server_cmd,  :out => w2, :in => r1)
-          p2 = spawn(ssh_cmd, :out => w1, :in => r2)
+          p1 = spawn(sftp_server_cmd, :out => w2, :in => r1, :err => t1)
```
Comment (hferentschik) [inline lib/vagrant-sshfs/cap/linux/sshfs_mount.rb]: I think you want to make sure that the new processes get a new process group. This only works OS specific though. See also - https://github.com/vagrant-landrush/landrush/blob/master/lib/landrush/server.rb#L71.

Not sure whether this has a direct effect here, but I several issues with the spawned processes until I properly made sure that they are completely unrelated (not even sharing the process group)

Comment (hferentschik) [inline lib/vagrant-sshfs/cap/linux/sshfs_mount.rb]: the process group flags don't seem to make a difference :-(

File: lib/vagrant-sshfs/cap/linux/sshfs_mount.rb (hunk 1 above)
Comment (dustymabe) [inline lib/vagrant-sshfs/cap/linux/sshfs_mount.rb]: #27, as reported stems from the fact that the STDERR stream was never closed and was still sending output to the terminal. Using a tmpfile instead means it is sending it somewhere else instead, and the installer doesn't get caught up on that fact.

**Pull request merged**